homeassistant/components/esphome/__init__.py

```diff
@@ -304,17 +305,67 @@ async def _register_service(
 ):
     service_name = f"{entry_data.device_info.name}_{service.name}"
     schema = {}
+    fields = {}
+
     for arg in service.args:
-        schema[vol.Required(arg.name)] = {
-            UserServiceArgType.BOOL: cv.boolean,
-            UserServiceArgType.INT: vol.Coerce(int),
-            UserServiceArgType.FLOAT: vol.Coerce(float),
-            UserServiceArgType.STRING: cv.string,
-            UserServiceArgType.BOOL_ARRAY: [cv.boolean],
-            UserServiceArgType.INT_ARRAY: [vol.Coerce(int)],
-            UserServiceArgType.FLOAT_ARRAY: [vol.Coerce(float)],
-            UserServiceArgType.STRING_ARRAY: [cv.string],
+        metadata = {
+            UserServiceArgType.BOOL: {
+                "validator": cv.boolean,
+                "description": "A boolean.",
```

Comment:
This is a bit lame for a description. Should we leave it out instead?

The same applies to all other occurances

Reply:
You mean with no description at all? We **only get** name and type information from ESPHome [here](https://esphome.io/components/api.html#user-defined-services). Adding a description field in ESPHome is doable, but that will require way more work, at least 4 prs involved. 

Anyway: Should it rely only on fill example data to infer type (when in yaml view)

Reply:
Boolean will be displayed as a toggle in the UI. Writing "A boolean." in the description is kinda useless. Maybe just leave it out / remove it. I don't see the added value here.

Reply:
> The same applies to all other occurances

what about list of Booleans, list of integers, etc. No UI selector for those so, should user rely on fill example data (not available on UI) this is going wrong IMO so removing type information from description defeats 50% intent of this PR.

Reply:
> what about list of Booleans, list of integers, etc

For those it makes sense, however, it doesn't say lists in those cases; but arrays (which is not a term used in the Home Assistant world).